dbt_platform_helper/utils/validation.py

```diff
@@ -456,6 +465,100 @@ def is_between(value):
 )
 
 
+def validate_platform_config(config):
+    get_aws_session_or_abort()  # Ensure we have a valid session as validation requires it.
+    PLATFORM_CONFIG_SCHEMA.validate(config)
+    _validate_environment_pipelines(config)
+    _validate_codebase_pipelines(config)
+
+
+def _validate_environment_pipelines(config):
+    enriched_config = apply_environment_defaults(config)
+    bad_pipelines = {}
+
+    for pipeline_name, pipeline in enriched_config.get("environment_pipelines", {}).items():
+        bad_envs = []
+        account = pipeline.get("account", None)
+        if account:
+            for env in pipeline.get("environments", {}).keys():
+                env_account = (
+                    enriched_config.get("environments", {})
+                    .get(env, {})
+                    .get("accounts", {})
+                    .get("deploy", {})
+                    .get("name")
+                )
+                if env_account and not env_account == account:
+                    bad_envs.append(env)
+        if bad_envs:
+            bad_pipelines[pipeline_name] = {"account": account, "bad_envs": bad_envs}
+    if bad_pipelines:
+        message = "The following pipelines are misconfigured:"
+        for pipeline, detail in bad_pipelines.items():
+            envs = detail["bad_envs"]
+            acc = detail["account"]
+            message += f"  '{pipeline}' - these environments are not in the '{acc}' account: {', '.join(envs)}\n"
+        abort_with_error(message)
+
+
+def _validate_codebase_pipelines(pipeline_config):
+    if not (CODEBASE_PIPELINES_KEY in pipeline_config or ENVIRONMENTS_KEY in pipeline_config):
```

Comment:
Wouldn't we need both of these?
```suggestion
    if not (CODEBASE_PIPELINES_KEY in pipeline_config and ENVIRONMENTS_KEY in pipeline_config):
```

Reply:
Not sure tbh. I just moved this function. It's existing functionality.

Reply:
Actually re-reading it, it's:
`not (A or B)` which is equivalent to `not A and not B` which is correct.

Reply:
It looks wrong to me, I think we need `CODEBASE_PIPELINES_KEY` and `ENVIRONMENTS_KEY` in the code that follows

Don't we want it to error if `not A or not B`?

Then we don't need the `if CODEBASE_PIPELINES_KEY in pipeline_config` which at the time of writing is on line 508.

Reply:
Not sure if you resolved this before my last comment, so unresolving.

Reply:
Pretty sure the current logic is sound. The command generates *copilot* pipelines. So it needs at least one of codebase_pipelines (where it gets codebase pipelines config from) or environments (where it gets environment pipeline config from) but not both.

Reply:
We have agreed verbally to remove this check as it should be possible to run `platform-helper generate` before you have pipelines configured.